### PR TITLE
Added default favicon and title

### LINKED ISSUE #20: Serve a default favicon.ico
A lot of web frameworks now a days serve a default favicon if the user does not provide one.

The benefit of this is simply not seeing 404 in the console as all browsers attempt to load it.

We should design some kind of logo and serve it as a default favicon.

### PR DESCRIPTION
**Pull request by ThomasJRyan**
Closes #20

### CODE REVIEW

File: src/rapidhtml/app.py [hunk 1]
```diff
@@ -20,15 +23,51 @@ class RapidHTML(Starlette):
...
+
+        @self.route("/favicon.ico")
+        async def favicon_route():
+            nonlocal favicon_path
+
+            if favicon_path is None:
+                return Response(get_default_favicon(), media_type="image/svg+xml")
+
+            with open(favicon_path, "rb") as f:
+                return FileResponse(f.read())
```
Comment (ThomasJRyan) [inline src/rapidhtml/app.py:69]: Thinking about it more, maybe I shouldn't put this inside the init 🤔 Maybe a "default routes" folder or something?

File: src/rapidhtml/app.py [hunk 2]
```diff
@@ -20,15 +23,51 @@ class RapidHTML(Starlette):
...
         if not html_head:
             html_head = ()
-        self.html_head = (Script(src="https://unpkg.com/htmx.org@2.0.1"),) + tuple(
-            html_head
-        )
+        self.html_head = [
+            Title(title),
+            Script(src="https://unpkg.com/htmx.org@2.0.1"),
+        ]
+        self.html_head.extend(html_head)
```
Comment (ThomasJRyan) [inline src/rapidhtml/app.py:56]: This PR should probably go in before significant work is done on https://github.com/ONTechNITS/RapidHTML/issues/38

File: tests/test_app.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    assert "<title>RapidHTML</title>" in response.text
+    assert "<h1>foobar</h1>" in response.text
+    assert "htmx.org" in response.text
+
+
+def test_favicon(client):
+    response = client.get("/favicon.ico")
+    assert response.status_code == 200
+    assert response.headers["content-type"] == "image/svg+xml"
+    assert response.content == get_default_favicon()
```
Comment (rashidnhm) [inline tests/test_app.py:40]: I am not sure myself, is this the best way to compare two large byte strings in python? :sweat_smile:

File: src/rapidhtml/static/RapidHTML.svg
[the remark this answers is not in the document]
Comment (ThomasJRyan) [reply src/rapidhtml/static/RapidHTML.svg:1]: Bird was a draft. The logo is still going through revisions. I'll want a proper svg of it before actually using it here. For now this silly R will be an acceptable placeholder

File: src/rapidhtml/app.py [hunk 4]
```diff
@@ -63,6 +83,16 @@ def __init__(
...
 
+        @self.route("/favicon.ico")
+        async def favicon_route():
+            nonlocal favicon_path
+
+            if favicon_path is None:
+                return Response(get_default_favicon(), media_type="image/svg+xml")
+
+            with open(favicon_path, "rb") as f:
+                return FileResponse(f.read())
```
[the remark this answers is not in the document]
Comment (ThomasJRyan) [reply src/rapidhtml/app.py]: Good point actually. Changed up how this is being done. Now the file data will be stored on the app object

File: tests/test_app.py (hunk 3 above)
Comment (ThomasJRyan) [reply tests/test_app.py:40]: Best way? Likely not... For the purposes of a unittest I think it works. Though I would like to know a better option for sure

I mostly just want to make sure the correct file is being passed. Could probably use a better test

**Pull request merged**